// @mantine/dates/esm/components/YearPicker/YearPicker.mjs
"use client";
import { useDatesState } from "../../hooks/use-dates-state/use-dates-state.mjs";
import { Calendar } from "../Calendar/Calendar.mjs";
import { jsx } from "react/jsx-runtime";
import { factory, useProps, useResolvedStylesApi } from "@mantine/core";
//#region packages/@mantine/dates/src/components/YearPicker/YearPicker.tsx
const defaultProps = { type: "default" };
const YearPicker = factory((_props) => {
	const props = useProps("YearPicker", defaultProps, _props);
	const { classNames, styles, vars, type, defaultValue, value, onChange, __staticSelector, getYearControlProps, allowSingleDateInRange, allowDeselect, onMouseLeave, onYearSelect, __updateDateOnYearSelect, ...others } = props;
	const { onDateChange, onRootMouseLeave, onHoveredDateChange, getControlProps } = useDatesState({
		type,
		level: "year",
		allowDeselect,
		allowSingleDateInRange,
		value,
		defaultValue,
		onChange,
		onMouseLeave
	});
	const { resolvedClassNames, resolvedStyles } = useResolvedStylesApi({
		classNames,
		styles,
		props
	});
	return /* @__PURE__ */ jsx(Calendar, {
		minLevel: "decade",
		__updateDateOnYearSelect: __updateDateOnYearSelect ?? false,
		__staticSelector: __staticSelector || "YearPicker",
		onMouseLeave: onRootMouseLeave,
		onYearMouseEnter: (_event, date) => onHoveredDateChange(date),
		onYearSelect: (date) => {
			onDateChange(date);
			onYearSelect?.(date);
		},
		getYearControlProps: (date) => ({
			...getControlProps(date),
			...getYearControlProps?.(date)
		}),
		classNames: resolvedClassNames,
		styles: resolvedStyles,
		...others
	});
});
YearPicker.classes = Calendar.classes;
YearPicker.displayName = "@mantine/dates/YearPicker";
//#endregion
export { YearPicker };
