// @mantine/dates/esm/components/TimePicker/use-time-picker.mjs
"use client";
import { clampTime } from "./utils/clamp-time/clamp-time.mjs";
import { getParsedTime } from "./utils/get-parsed-time/get-parsed-time.mjs";
import { getTimeString } from "./utils/get-time-string/get-time-string.mjs";
import { useRef, useState } from "react";
import { useDidUpdate } from "@mantine/hooks";
//#region packages/@mantine/dates/src/components/TimePicker/use-time-picker.ts
function useTimePicker({ value, defaultValue, onChange, format, amPmLabels, withSeconds = false, min, max, clearable, readOnly, disabled, pasteSplit }) {
	const parsedTime = getParsedTime({
		time: value || defaultValue || "",
		amPmLabels,
		format
	});
	const initialTimeString = getTimeString({
		hours: parsedTime.hours,
		minutes: parsedTime.minutes,
		seconds: parsedTime.seconds,
		format,
		withSeconds,
		amPm: parsedTime.amPm,
		amPmLabels
	});
	const acceptChange = useRef(true);
	const wasInvalidBefore = useRef(!initialTimeString.valid);
	const [hours, setHours] = useState(parsedTime.hours);
	const [minutes, setMinutes] = useState(parsedTime.minutes);
	const [seconds, setSeconds] = useState(parsedTime.seconds);
	const [amPm, setAmPm] = useState(parsedTime.amPm);
	const isClearable = clearable && !readOnly && !disabled && (hours !== null || minutes !== null || seconds !== null || amPm !== null);
	const hoursRef = useRef(null);
	const minutesRef = useRef(null);
	const secondsRef = useRef(null);
	const amPmRef = useRef(null);
	const focus = (field) => {
		if (field === "hours") hoursRef.current?.focus();
		if (field === "minutes") minutesRef.current?.focus();
		if (field === "seconds") secondsRef.current?.focus();
		if (field === "amPm") amPmRef.current?.focus();
	};
	const handleTimeChange = (field, val) => {
		const timeString = getTimeString({
			hours,
			minutes,
			seconds,
			amPm,
			[field]: val,
			format,
			withSeconds,
			amPmLabels
		});
		if (timeString.valid) {
			acceptChange.current = false;
			wasInvalidBefore.current = false;
			if (field === "hours") setHours(val);
			if (field === "minutes") setMinutes(val);
			if (field === "seconds") setSeconds(val);
			if (field === "amPm") setAmPm(val);
			onChange?.(timeString.value);
		} else {
			acceptChange.current = false;
			if (!wasInvalidBefore.current) {
				onChange?.("");
				wasInvalidBefore.current = true;
			}
		}
	};
	const setTimeString = (timeString) => {
		acceptChange.current = false;
		const parsedTime = getParsedTime({
			time: timeString,
			amPmLabels,
			format
		});
		setHours(parsedTime.hours);
		setMinutes(parsedTime.minutes);
		setSeconds(parsedTime.seconds);
		setAmPm(parsedTime.amPm);
		wasInvalidBefore.current = !getTimeString({
			...parsedTime,
			format,
			withSeconds,
			amPmLabels
		}).valid;
		onChange?.(timeString);
	};
	const onHoursChange = (value) => {
		let adjustedValue = value;
		if (format === "12h" && typeof value === "number" && value > 12) adjustedValue = (value - 1) % 12 + 1;
		setHours(adjustedValue);
		handleTimeChange("hours", adjustedValue);
		focus("hours");
	};
	const onMinutesChange = (value) => {
		setMinutes(value);
		handleTimeChange("minutes", value);
		focus("minutes");
	};
	const onSecondsChange = (value) => {
		setSeconds(value);
		handleTimeChange("seconds", value);
		focus("seconds");
	};
	const onAmPmChange = (value) => {
		setAmPm(value);
		handleTimeChange("amPm", value);
		focus("amPm");
	};
	const clear = () => {
		acceptChange.current = false;
		setHours(null);
		setMinutes(null);
		setSeconds(null);
		setAmPm(null);
		onChange?.("");
		wasInvalidBefore.current = true;
		focus("hours");
	};
	const onPaste = (event) => {
		event.preventDefault();
		const pastedValue = event.clipboardData.getData("text");
		const parsedTime = (pasteSplit || getParsedTime)({
			time: pastedValue,
			format,
			amPmLabels
		});
		const timeString = getTimeString({
			...parsedTime,
			format,
			withSeconds,
			amPmLabels
		});
		if (timeString.valid) {
			acceptChange.current = false;
			const clamped = clampTime(timeString.value, min || "00:00:00", max || "23:59:59");
			onChange?.(clamped.timeString);
			setHours(parsedTime.hours);
			setMinutes(parsedTime.minutes);
			setSeconds(parsedTime.seconds);
			setAmPm(parsedTime.amPm);
		}
	};
	const hiddenInputValue = getTimeString({
		hours,
		minutes,
		seconds,
		format,
		withSeconds,
		amPm,
		amPmLabels
	});
	useDidUpdate(() => {
		if (value === "") {
			acceptChange.current = false;
			setHours(null);
			setMinutes(null);
			setSeconds(null);
			setAmPm(null);
			acceptChange.current = true;
			return;
		}
		if (acceptChange.current && typeof value === "string") {
			const parsedTime = getParsedTime({
				time: value || "",
				amPmLabels,
				format
			});
			setHours(parsedTime.hours);
			setMinutes(parsedTime.minutes);
			setSeconds(parsedTime.seconds);
			setAmPm(parsedTime.amPm);
		}
		acceptChange.current = true;
	}, [value]);
	return {
		refs: {
			hours: hoursRef,
			minutes: minutesRef,
			seconds: secondsRef,
			amPm: amPmRef
		},
		values: {
			hours,
			minutes,
			seconds,
			amPm
		},
		setHours: onHoursChange,
		setMinutes: onMinutesChange,
		setSeconds: onSecondsChange,
		setAmPm: onAmPmChange,
		focus,
		clear,
		onPaste,
		setTimeString,
		isClearable,
		hiddenInputValue: hiddenInputValue.value
	};
}
//#endregion
export { useTimePicker };
