// @mantine/dates/cjs/components/Day/Day.cjs
"use client";
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_Day_module = require("./Day.module.cjs");
let dayjs = require("dayjs");
dayjs = require_runtime.__toESM(dayjs);
let react_jsx_runtime = require("react/jsx-runtime");
let _mantine_core = require("@mantine/core");
//#region packages/@mantine/dates/src/components/Day/Day.tsx
const varsResolver = (0, _mantine_core.createVarsResolver)((_, { size }) => ({ day: { "--day-size": (0, _mantine_core.getSize)(size, "day-size") } }));
const Day = (0, _mantine_core.factory)((_props) => {
	const props = (0, _mantine_core.useProps)("Day", null, _props);
	const { classNames, className, style, styles, unstyled, vars, date, disabled, __staticSelector, weekend, outside, selected, renderDay, inRange, firstInRange, lastInRange, hidden, static: isStatic, highlightToday, fullWidth, attributes, ...others } = props;
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(_mantine_core.UnstyledButton, {
		...(0, _mantine_core.useStyles)({
			name: __staticSelector || "Day",
			classes: require_Day_module.default,
			props,
			className,
			style,
			classNames,
			styles,
			unstyled,
			attributes,
			vars,
			varsResolver,
			rootSelector: "day"
		})("day", { style: hidden ? { display: "none" } : void 0 }),
		component: isStatic ? "div" : "button",
		disabled,
		"data-today": (0, dayjs.default)(date).isSame(/* @__PURE__ */ new Date(), "day") || void 0,
		"data-hidden": hidden || void 0,
		"data-highlight-today": highlightToday || void 0,
		"data-disabled": disabled || void 0,
		"data-weekend": !disabled && !outside && weekend || void 0,
		"data-outside": !disabled && outside || void 0,
		"data-selected": !disabled && selected || void 0,
		"data-in-range": inRange && !disabled || void 0,
		"data-first-in-range": firstInRange && !disabled || void 0,
		"data-last-in-range": lastInRange && !disabled || void 0,
		"data-static": isStatic || void 0,
		"data-full-width": fullWidth || void 0,
		unstyled,
		...others,
		children: renderDay?.(date) || (0, dayjs.default)(date).date()
	});
});
Day.classes = require_Day_module.default;
Day.varsResolver = varsResolver;
Day.displayName = "@mantine/dates/Day";
//#endregion
exports.Day = Day;
