// @mantine/dates/cjs/components/DateTimePicker/DateTimePicker.cjs
"use client";
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_assign_time = require("../../utils/assign-time/assign-time.cjs");
const require_get_default_clamped_date = require("../../utils/get-default-clamped-date/get-default-clamped-date.cjs");
const require_clamp_date = require("../../utils/clamp-date/clamp-date.cjs");
const require_use_dates_context = require("../DatesProvider/use-dates-context.cjs");
const require_TimePicker = require("../TimePicker/TimePicker.cjs");
const require_PickerInputBase = require("../PickerInputBase/PickerInputBase.cjs");
const require_use_uncontrolled_dates = require("../../hooks/use-uncontrolled-dates/use-uncontrolled-dates.cjs");
const require_pick_calendar_levels_props = require("../Calendar/pick-calendar-levels-props/pick-calendar-levels-props.cjs");
const require_DatePicker = require("../DatePicker/DatePicker.cjs");
const require_get_min_max_time = require("./get-min-max-time/get-min-max-time.cjs");
const require_DateTimePicker_module = require("./DateTimePicker.module.cjs");
let dayjs = require("dayjs");
dayjs = require_runtime.__toESM(dayjs);
let react = require("react");
let react_jsx_runtime = require("react/jsx-runtime");
let _mantine_core = require("@mantine/core");
let _mantine_hooks = require("@mantine/hooks");
//#region packages/@mantine/dates/src/components/DateTimePicker/DateTimePicker.tsx
const defaultProps = {
	dropdownType: "popover",
	size: "sm"
};
const DateTimePicker = (0, _mantine_core.factory)((_props) => {
	const props = (0, _mantine_core.useProps)("DateTimePicker", defaultProps, _props);
	const { value, defaultValue, onChange, valueFormat, locale, classNames, styles, unstyled, timePickerProps, submitButtonProps, withSeconds, level, defaultLevel, size, variant, dropdownType, vars, minDate, maxDate, defaultDate, defaultTimeValue, presets, attributes, onDropdownClose, ...rest } = props;
	const getStyles = (0, _mantine_core.useStyles)({
		name: "DateTimePicker",
		classes: require_DateTimePicker_module.default,
		props,
		classNames,
		styles,
		unstyled,
		attributes,
		vars
	});
	const { resolvedClassNames, resolvedStyles } = (0, _mantine_core.useResolvedStylesApi)({
		classNames,
		styles,
		props
	});
	const _withSeconds = withSeconds || timePickerProps?.withSeconds;
	const _valueFormat = valueFormat || (_withSeconds ? "DD/MM/YYYY HH:mm:ss" : "DD/MM/YYYY HH:mm");
	const timePickerRef = (0, react.useRef)(null);
	const timePickerRefMerged = (0, _mantine_hooks.useMergedRef)(timePickerRef, timePickerProps?.hoursRef);
	const { calendarProps: { allowSingleDateInRange, ...calendarProps }, others } = require_pick_calendar_levels_props.pickCalendarProps(rest);
	const ctx = require_use_dates_context.useDatesContext();
	const [_value, setValue] = require_use_uncontrolled_dates.useUncontrolledDates({
		type: "default",
		value,
		defaultValue,
		onChange,
		withTime: true
	});
	const _defaultDate = defaultDate || _value;
	const formatTime = (dateValue) => dateValue ? (0, dayjs.default)(dateValue).format(_withSeconds ? "HH:mm:ss" : "HH:mm") : "";
	const [timeValue, setTimeValue] = (0, react.useState)(defaultTimeValue || formatTime(_value));
	const [currentLevel, setCurrentLevel] = (0, react.useState)(level || defaultLevel || "month");
	const [dropdownOpened, dropdownHandlers] = (0, _mantine_hooks.useDisclosure)(false);
	const formattedValue = _value ? (0, dayjs.default)(_value).locale(ctx.getLocale(locale)).format(_valueFormat) : "";
	const handleTimeChange = (timeString) => {
		timePickerProps?.onChange?.(timeString);
		setTimeValue(timeString);
		if (timeString) setValue(require_assign_time.assignTime(_value, timeString));
	};
	const handleDateChange = (date) => {
		if (date) setValue(require_assign_time.assignTime(require_clamp_date.clampDate(minDate, maxDate, date), timeValue || defaultTimeValue || ""));
		timePickerRef.current?.focus();
	};
	const handleTimeInputKeyDown = (event) => {
		if (event.key === "Enter") {
			event.preventDefault();
			dropdownHandlers.close();
		}
	};
	(0, _mantine_hooks.useDidUpdate)(() => {
		if (!dropdownOpened) setTimeValue(formatTime(_value));
	}, [_value, dropdownOpened]);
	(0, _mantine_hooks.useDidUpdate)(() => {
		if (dropdownOpened) setCurrentLevel("month");
	}, [dropdownOpened]);
	const __stopPropagation = dropdownType === "popover";
	const handleDropdownClose = () => {
		const clamped = require_clamp_date.clampDate(minDate, maxDate, _value);
		if (_value && _value !== clamped) setValue(require_clamp_date.clampDate(minDate, maxDate, _value));
		onDropdownClose?.();
	};
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_PickerInputBase.PickerInputBase, {
		formattedValue,
		dropdownOpened: !rest.disabled ? dropdownOpened : false,
		dropdownHandlers,
		classNames: resolvedClassNames,
		styles: resolvedStyles,
		unstyled,
		onClear: () => setValue(null),
		shouldClear: !!_value,
		value: _value,
		size,
		variant,
		dropdownType,
		...others,
		type: "default",
		__staticSelector: "DateTimePicker",
		onDropdownClose: handleDropdownClose,
		withTime: true,
		attributes,
		children: [/* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_DatePicker.DatePicker, {
			...calendarProps,
			maxDate,
			minDate,
			size,
			variant,
			type: "default",
			value: _value,
			defaultDate: _defaultDate || require_get_default_clamped_date.getDefaultClampedDate({
				maxDate,
				minDate
			}),
			onChange: handleDateChange,
			locale,
			classNames: resolvedClassNames,
			styles: resolvedStyles,
			unstyled,
			__staticSelector: "DateTimePicker",
			__stopPropagation,
			level,
			defaultLevel,
			onLevelChange: (_level) => {
				setCurrentLevel(_level);
				calendarProps.onLevelChange?.(_level);
			},
			presets,
			__onPresetSelect: (val) => {
				setValue(val);
				val && setTimeValue(formatTime(val));
			},
			attributes
		}), currentLevel === "month" && /* @__PURE__ */ (0, react_jsx_runtime.jsxs)("div", {
			...getStyles("timeWrapper"),
			children: [/* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_TimePicker.TimePicker, {
				value: timeValue,
				withSeconds,
				unstyled,
				min: require_get_min_max_time.getMinTime({
					minDate,
					value: _value
				}),
				max: require_get_min_max_time.getMaxTime({
					maxDate,
					value: _value
				}),
				...timePickerProps,
				...getStyles("timeInput", {
					className: timePickerProps?.className,
					style: timePickerProps?.style
				}),
				onChange: handleTimeChange,
				onKeyDown: handleTimeInputKeyDown,
				size,
				"data-mantine-stop-propagation": __stopPropagation || void 0,
				hoursRef: timePickerRefMerged,
				attributes
			}), /* @__PURE__ */ (0, react_jsx_runtime.jsx)(_mantine_core.ActionIcon, {
				variant: "default",
				size: `input-${size || "sm"}`,
				...getStyles("submitButton", {
					className: submitButtonProps?.className,
					style: submitButtonProps?.style
				}),
				unstyled,
				"data-mantine-stop-propagation": __stopPropagation || void 0,
				children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(_mantine_core.CheckIcon, { size: "30%" }),
				...submitButtonProps,
				onClick: (event) => {
					submitButtonProps?.onClick?.(event);
					dropdownHandlers.close();
					handleDropdownClose();
				}
			})]
		})]
	});
});
DateTimePicker.classes = {
	...require_DateTimePicker_module.default,
	...require_PickerInputBase.PickerInputBase.classes,
	...require_DatePicker.DatePicker.classes
};
DateTimePicker.displayName = "@mantine/dates/DateTimePicker";
//#endregion
exports.DateTimePicker = DateTimePicker;
