// @mantine/dates/cjs/components/TimePicker/TimePresets/TimePresetGroup.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_TimePicker_context = require("../TimePicker.context.cjs");
const require_is_same_time = require("../utils/is-same-time/is-same-time.cjs");
const require_TimePresetControl = require("./TimePresetControl.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
let _mantine_core = require("@mantine/core");
//#region packages/@mantine/dates/src/components/TimePicker/TimePresets/TimePresetGroup.tsx
function TimePresetGroup({ value, data, onChange, format, amPmLabels, withSeconds }) {
	const ctx = require_TimePicker_context.useTimePickerContext();
	const items = data.values.map((item) => /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_TimePresetControl.TimePresetControl, {
		value: item,
		format,
		amPmLabels,
		withSeconds,
		active: require_is_same_time.isSameTime({
			time: item,
			compare: value,
			withSeconds
		}),
		onChange
	}, item));
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)("div", {
		...ctx.getStyles("presetsGroup"),
		children: [/* @__PURE__ */ (0, react_jsx_runtime.jsx)("div", {
			...ctx.getStyles("presetsGroupLabel"),
			children: data.label
		}), /* @__PURE__ */ (0, react_jsx_runtime.jsx)(_mantine_core.SimpleGrid, {
			cols: withSeconds ? 2 : 3,
			spacing: 4,
			children: items
		})]
	});
}
TimePresetGroup.displayName = "@mantine/dates/TimePresetGroup";
//#endregion
exports.TimePresetGroup = TimePresetGroup;
