// @mantine/dates/cjs/components/WeekdaysRow/WeekdaysRow.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_use_dates_context = require("../DatesProvider/use-dates-context.cjs");
const require_get_weekdays_names = require("./get-weekdays-names/get-weekdays-names.cjs");
const require_WeekdaysRow_module = require("./WeekdaysRow.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
let _mantine_core = require("@mantine/core");
//#region packages/@mantine/dates/src/components/WeekdaysRow/WeekdaysRow.tsx
const varsResolver = (0, _mantine_core.createVarsResolver)((_, { size }) => ({ weekdaysRow: {
	"--wr-fz": (0, _mantine_core.getFontSize)(size),
	"--wr-spacing": (0, _mantine_core.getSpacing)(size)
} }));
const WeekdaysRow = (0, _mantine_core.factory)((_props) => {
	const props = (0, _mantine_core.useProps)("WeekdaysRow", null, _props);
	const { classNames, className, style, styles, unstyled, vars, locale, firstDayOfWeek, weekdayFormat, cellComponent: CellComponent = "th", __staticSelector, withWeekNumbers, attributes, ...others } = props;
	const getStyles = (0, _mantine_core.useStyles)({
		name: __staticSelector || "WeekdaysRow",
		classes: require_WeekdaysRow_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver,
		rootSelector: "weekdaysRow"
	});
	const ctx = require_use_dates_context.useDatesContext();
	const weekdays = require_get_weekdays_names.getWeekdayNames({
		locale: ctx.getLocale(locale),
		format: weekdayFormat,
		firstDayOfWeek: ctx.getFirstDayOfWeek(firstDayOfWeek)
	}).map((weekday, index) => /* @__PURE__ */ (0, react_jsx_runtime.jsx)(CellComponent, {
		...getStyles("weekday"),
		children: weekday
	}, index));
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(_mantine_core.Box, {
		component: "tr",
		...getStyles("weekdaysRow"),
		...others,
		children: [withWeekNumbers && /* @__PURE__ */ (0, react_jsx_runtime.jsx)(CellComponent, {
			...getStyles("weekday"),
			children: "#"
		}), weekdays]
	});
});
WeekdaysRow.classes = require_WeekdaysRow_module.default;
WeekdaysRow.varsResolver = varsResolver;
WeekdaysRow.displayName = "@mantine/dates/WeekdaysRow";
//#endregion
exports.WeekdaysRow = WeekdaysRow;
