// @mantine/dates/cjs/components/MonthPicker/MonthPicker.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_use_dates_state = require("../../hooks/use-dates-state/use-dates-state.cjs");
const require_Calendar = require("../Calendar/Calendar.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
let _mantine_core = require("@mantine/core");
//#region packages/@mantine/dates/src/components/MonthPicker/MonthPicker.tsx
const defaultProps = { type: "default" };
const MonthPicker = (0, _mantine_core.factory)((_props) => {
	const props = (0, _mantine_core.useProps)("MonthPicker", defaultProps, _props);
	const { classNames, styles, vars, type, defaultValue, value, onChange, __staticSelector, getMonthControlProps, allowSingleDateInRange, allowDeselect, onMouseLeave, onMonthSelect, __updateDateOnMonthSelect, onLevelChange, ...others } = props;
	const { onDateChange, onRootMouseLeave, onHoveredDateChange, getControlProps } = require_use_dates_state.useDatesState({
		type,
		level: "month",
		allowDeselect,
		allowSingleDateInRange,
		value,
		defaultValue,
		onChange,
		onMouseLeave
	});
	const { resolvedClassNames, resolvedStyles } = (0, _mantine_core.useResolvedStylesApi)({
		classNames,
		styles,
		props
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Calendar.Calendar, {
		minLevel: "year",
		__updateDateOnMonthSelect: __updateDateOnMonthSelect ?? false,
		__staticSelector: __staticSelector || "MonthPicker",
		onMouseLeave: onRootMouseLeave,
		onMonthMouseEnter: (_event, date) => onHoveredDateChange(date),
		onMonthSelect: (date) => {
			onDateChange(date);
			onMonthSelect?.(date);
		},
		getMonthControlProps: (date) => ({
			...getControlProps(date),
			...getMonthControlProps?.(date)
		}),
		classNames: resolvedClassNames,
		styles: resolvedStyles,
		onLevelChange,
		...others
	});
});
MonthPicker.classes = require_Calendar.Calendar.classes;
MonthPicker.displayName = "@mantine/dates/MonthPicker";
//#endregion
exports.MonthPicker = MonthPicker;
