// @mantine/dates/esm/components/Day/Day.mjs
"use client";
import Day_module_default from "./Day.module.mjs";
import dayjs from "dayjs";
import { jsx } from "react/jsx-runtime";
import { UnstyledButton, createVarsResolver, factory, getSize, useProps, useStyles } from "@mantine/core";
//#region packages/@mantine/dates/src/components/Day/Day.tsx
const varsResolver = createVarsResolver((_, { size }) => ({ day: { "--day-size": getSize(size, "day-size") } }));
const Day = factory((_props) => {
	const props = useProps("Day", null, _props);
	const { classNames, className, style, styles, unstyled, vars, date, disabled, __staticSelector, weekend, outside, selected, renderDay, inRange, firstInRange, lastInRange, hidden, static: isStatic, highlightToday, fullWidth, attributes, ...others } = props;
	return /* @__PURE__ */ jsx(UnstyledButton, {
		...useStyles({
			name: __staticSelector || "Day",
			classes: Day_module_default,
			props,
			className,
			style,
			classNames,
			styles,
			unstyled,
			attributes,
			vars,
			varsResolver,
			rootSelector: "day"
		})("day", { style: hidden ? { display: "none" } : void 0 }),
		component: isStatic ? "div" : "button",
		disabled,
		"data-today": dayjs(date).isSame(/* @__PURE__ */ new Date(), "day") || void 0,
		"data-hidden": hidden || void 0,
		"data-highlight-today": highlightToday || void 0,
		"data-disabled": disabled || void 0,
		"data-weekend": !disabled && !outside && weekend || void 0,
		"data-outside": !disabled && outside || void 0,
		"data-selected": !disabled && selected || void 0,
		"data-in-range": inRange && !disabled || void 0,
		"data-first-in-range": firstInRange && !disabled || void 0,
		"data-last-in-range": lastInRange && !disabled || void 0,
		"data-static": isStatic || void 0,
		"data-full-width": fullWidth || void 0,
		unstyled,
		...others,
		children: renderDay?.(date) || dayjs(date).date()
	});
});
Day.classes = Day_module_default;
Day.varsResolver = varsResolver;
Day.displayName = "@mantine/dates/Day";
//#endregion
export { Day };
