// @mantine/dates/cjs/components/YearPicker/YearPicker.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_use_dates_state = require("../../hooks/use-dates-state/use-dates-state.cjs");
const require_Calendar = require("../Calendar/Calendar.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
let _mantine_core = require("@mantine/core");
//#region packages/@mantine/dates/src/components/YearPicker/YearPicker.tsx
const defaultProps = { type: "default" };
const YearPicker = (0, _mantine_core.factory)((_props) => {
	const props = (0, _mantine_core.useProps)("YearPicker", defaultProps, _props);
	const { classNames, styles, vars, type, defaultValue, value, onChange, __staticSelector, getYearControlProps, allowSingleDateInRange, allowDeselect, onMouseLeave, onYearSelect, __updateDateOnYearSelect, ...others } = props;
	const { onDateChange, onRootMouseLeave, onHoveredDateChange, getControlProps } = require_use_dates_state.useDatesState({
		type,
		level: "year",
		allowDeselect,
		allowSingleDateInRange,
		value,
		defaultValue,
		onChange,
		onMouseLeave
	});
	const { resolvedClassNames, resolvedStyles } = (0, _mantine_core.useResolvedStylesApi)({
		classNames,
		styles,
		props
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Calendar.Calendar, {
		minLevel: "decade",
		__updateDateOnYearSelect: __updateDateOnYearSelect ?? false,
		__staticSelector: __staticSelector || "YearPicker",
		onMouseLeave: onRootMouseLeave,
		onYearMouseEnter: (_event, date) => onHoveredDateChange(date),
		onYearSelect: (date) => {
			onDateChange(date);
			onYearSelect?.(date);
		},
		getYearControlProps: (date) => ({
			...getControlProps(date),
			...getYearControlProps?.(date)
		}),
		classNames: resolvedClassNames,
		styles: resolvedStyles,
		...others
	});
});
YearPicker.classes = require_Calendar.Calendar.classes;
YearPicker.displayName = "@mantine/dates/YearPicker";
//#endregion
exports.YearPicker = YearPicker;
