// @mantine/dates/cjs/components/TimePicker/utils/get-parsed-time/get-parsed-time.cjs
"use client";
const require_split_time_string = require("../split-time-string/split-time-string.cjs");
//#region packages/@mantine/dates/src/components/TimePicker/utils/get-parsed-time/get-parsed-time.ts
function convertTimeTo12HourFormat({ hours, minutes, seconds, amPmLabels }) {
	if (hours === null) return {
		hours: null,
		minutes: null,
		seconds: null,
		amPm: null
	};
	const amPm = hours >= 12 ? amPmLabels.pm : amPmLabels.am;
	return {
		hours: hours % 12 === 0 ? 12 : hours % 12,
		minutes: typeof minutes === "number" ? minutes : null,
		seconds: typeof seconds === "number" ? seconds : null,
		amPm
	};
}
function getParsedTime({ time, format, amPmLabels }) {
	if (time === "") return {
		hours: null,
		minutes: null,
		seconds: null,
		amPm: null
	};
	const { hours, minutes, seconds } = require_split_time_string.splitTimeString(time);
	const parsed = {
		hours,
		minutes,
		seconds
	};
	if (format === "12h") return convertTimeTo12HourFormat({
		...parsed,
		amPmLabels
	});
	return {
		...parsed,
		amPm: null
	};
}
//#endregion
exports.getParsedTime = getParsedTime;
