// @mantine/dates/cjs/components/MonthPickerInput/MonthPickerInput.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_get_default_clamped_date = require("../../utils/get-default-clamped-date/get-default-clamped-date.cjs");
const require_PickerInputBase = require("../PickerInputBase/PickerInputBase.cjs");
const require_use_dates_input = require("../../hooks/use-dates-input/use-dates-input.cjs");
const require_pick_calendar_levels_props = require("../Calendar/pick-calendar-levels-props/pick-calendar-levels-props.cjs");
const require_MonthPicker = require("../MonthPicker/MonthPicker.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
let _mantine_core = require("@mantine/core");
//#region packages/@mantine/dates/src/components/MonthPickerInput/MonthPickerInput.tsx
const defaultProps = {
	type: "default",
	size: "sm",
	valueFormat: "MMMM YYYY",
	closeOnChange: true,
	sortDates: true,
	dropdownType: "popover"
};
const MonthPickerInput = (0, _mantine_core.factory)((_props) => {
	const props = (0, _mantine_core.useProps)("MonthPickerInput", defaultProps, _props);
	const { type, value, defaultValue, onChange, valueFormat, labelSeparator, locale, classNames, styles, unstyled, closeOnChange, size, variant, dropdownType, sortDates, minDate, maxDate, vars, valueFormatter, attributes, ...rest } = props;
	const { resolvedClassNames, resolvedStyles } = (0, _mantine_core.useResolvedStylesApi)({
		classNames,
		styles,
		props
	});
	const { calendarProps, others } = require_pick_calendar_levels_props.pickCalendarProps(rest);
	const { _value, setValue, formattedValue, dropdownHandlers, dropdownOpened, onClear, shouldClear } = require_use_dates_input.useDatesInput({
		type,
		value,
		defaultValue,
		onChange,
		locale,
		format: valueFormat,
		labelSeparator,
		closeOnChange,
		sortDates,
		valueFormatter
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_PickerInputBase.PickerInputBase, {
		formattedValue,
		dropdownOpened,
		dropdownHandlers,
		classNames: resolvedClassNames,
		styles: resolvedStyles,
		unstyled,
		onClear,
		shouldClear,
		value: _value,
		size,
		variant,
		dropdownType,
		...others,
		attributes,
		type,
		__staticSelector: "MonthPickerInput",
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_MonthPicker.MonthPicker, {
			...calendarProps,
			size,
			variant,
			type,
			value: _value,
			defaultDate: calendarProps.defaultDate || (Array.isArray(_value) ? _value[0] || require_get_default_clamped_date.getDefaultClampedDate({
				maxDate,
				minDate
			}) : _value || require_get_default_clamped_date.getDefaultClampedDate({
				maxDate,
				minDate
			})),
			onChange: setValue,
			locale,
			classNames: resolvedClassNames,
			styles: resolvedStyles,
			unstyled,
			__staticSelector: "MonthPickerInput",
			__stopPropagation: dropdownType === "popover",
			minDate,
			maxDate,
			attributes
		})
	});
});
MonthPickerInput.classes = {
	...require_PickerInputBase.PickerInputBase.classes,
	...require_MonthPicker.MonthPicker.classes
};
MonthPickerInput.displayName = "@mantine/dates/MonthPickerInput";
//#endregion
exports.MonthPickerInput = MonthPickerInput;
